sympy/series/limits.py

```diff
@@ -146,15 +146,14 @@ def doit(self, **hints):
 
         if e.is_Mul:
             if abs(z0) is S.Infinity:
-                # XXX todo: this should probably be stated in the
-                # negative -- i.e. to exclude expressions that should
-                # not be handled this way but I'm not sure what that
-                # condition is; when ok is True it means that the leading
-                # term approach is going to succeed (hopefully)
+                ztemp = Symbol("z",real = True)
```

Comment:
@jksuom , you are right that the logarithm is calculated even before I substitute with `ztemp` but the answer is not `absolute`
Here is an example
```
x = Symbol("x")
log(exp(x))
returns log(exp(x)) i.e unevaluated.
```
bur if I do:
```
y = Symbol("y", real = True)
log(exp(x)).subs(x,y)
returns y
```
This is the reason I have defined the temporary variable.